src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1860,51 +1854,51 @@ fn handle_variant(
     if extra_fragment.is_some() {
         return Err(ErrorKind::AnchorFailure(AnchorFailure::RustdocAnchorConflict(res)));
     }
-    let parent = if let Some(parent) = cx.tcx.parent(res.def_id()) {
-        parent
-    } else {
-        return Err(ResolutionFailure::NoParentItem.into());
-    };
-    let parent_def = Res::Def(DefKind::Enum, parent);
-    let variant = cx.tcx.expect_variant_res(res);
-    Ok((parent_def, Some(format!("variant.{}", variant.ident.name))))
+    cx.tcx.parent(res.def_id()).map_or_else(
+        || Err(ResolutionFailure::NoParentItem.into()),
+        |parent| {
+            let parent_def = Res::Def(DefKind::Enum, parent);
+            let variant = cx.tcx.expect_variant_res(res);
+            Ok((parent_def, Some(format!("variant.{}", variant.ident.name))))
+        },
```

Comment:
I think you missed the last comment about this ;)
```suggestion
    cx.tcx.parent(res.def_id()).ok_or_else(
        || ResolutionFailure::NoParentItem.into(),
        |parent| {
            let parent_def = Res::Def(DefKind::Enum, parent);
            let variant = cx.tcx.expect_variant_res(res);
            (parent_def, Some(format!("variant.{}", variant.ident.name)))
        },
```

Reply:
Indeed, I have, thanks!

Reply:
I have a love-hate relationship with the result/option API, this change broke the build.

Reply:
@jyn514 let's wait with bors while I fix this :)